ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Serialisation/SerialiseNodeToClient.hs

```diff
@@ -173,22 +172,36 @@ instance SerialiseHFC xs
   Queries
 -------------------------------------------------------------------------------}
 
+querySupportedVersion :: Some (QueryHardFork xs)
```

Comment:
This is clean, nice. I do wonder if (if only for consistency with other encode functions that I'm recalling) it'd be better to leave the guard near the encoder.

You could get the same re-use in the case expression by pulling the `| otherwise ->` alternative out as a default `_otherwise` alternative like you currently have as a guard. Does that suggestion make sense? What you think? I'm regretting the loss of the locality of putting the guard on the encoder definition itself.